sushy_oem_idrac/resources/manager/manager.py

```diff
@@ -83,6 +102,90 @@ class DellManagerExtension(oem_base.OEMResourceBase):
     def import_system_configuration_uri(self):
         return self._actions.import_system_configuration.target_uri
 
+    @property
+    def job_service_uri(self):
+        return self._links.job_service.resource_uri
+
+    @property
+    def get_jobs_uri(self):
+        return self._links.jobs.resource_uri
+
+    @property
+    def get_remote_service_api_uri(self):
+        return ('%s%s' %
+                (self._links.lc_service.resource_uri,
+                self.REMOTE_API_STATUS_URI))
+
+    def clear_job_queue(self, job_ids=['JID_CLEARALL']):
+
+        if job_ids is None:
+            return None
+
+        delete_job_queue_uri = '%s%s' % (self.job_service_uri, self.CLEAR_JOBS_URI)
+        for job_id in job_ids:
+            payload = {'JobID':job_id}
+
+            try:
+                delete_job_response = asynchronous.http_call(
+                        self._conn, 'post',
+                        delete_job_queue_uri,
+                        headers = self.HEADERS,
+                        data = payload,
+                        verify = False)
+                return delete_job_response
+
+            except (sushy.exceptions.ServerSideError,
+                    sushy.exceptions.BadRequestError) as exc:
+                LOG.error('Dell OEM clear job queue failed, %s', err_msg)
+                raise sushy.exceptions.ServerSideError(error=exc)
+
+    def reset_idrac(self,manager=None):
+        payload = {"ResetType":"GracefulRestart"}
+
+        try:
+            reset_job_response = asynchronous.http_call(
+                            self._conn, 'post',
+                            manager._actions.reset.target_uri,
+                            data=(payload))
+
+        except (sushy.exceptions.ServerSideError,
+                sushy.exceptions.BadRequestError) as exc:
+            LOG.error('Dell OEM reset idrac failed, Reason : %s', exc)
+            raise sushy.exceptions.ServerSideError(error=exc)
+
```

Comment:
I think we should keep the logic of detecting when the reset is complete in the iDRAC driver on the ironic side, so the code below should be moved into management.py in the drac driver.

Reply:
Actually, let's keep this symmetrical with how python-dracclient is implemented and put the logic to detect when the reset is complete here.

Reply:
We need to add the code that detects when the iDRAC reset is complete here.  The algorithm is:
1. ping the iDRAC until it is not pingable
2. ping the iDRAC until it is pingable
3. execute GetRemoteServicesAPIStatus against the iDRAC, ignoring 400 errors until LCStatus=="Ready"
See: https://github.com/openstack/python-dracclient/blob/master/dracclient/client.py#L265
If you just immediately call GetRemoteServiceAPIStatus against the iDRAC after the reset, then it may return "Ready" immediately because it has just begun to shut down.
See: CES-11028
You can copy the _wait_for_host_state method directly from python-dracclient and add it to utils.py, then call it from here the same way we do in python-dracclient.

Reply:
Addressed as per python-dracclient implementation after reset completion.
Also added below methods.

1. wait_for_host_state method in ironic side's utils.py

2. wait_until_idrac_ready method in sushy-oem side as it contains idrac requests'/uri call for get remote status.